packages/react-aria-components/src/OverlayArrow.tsx

```diff
@@ -48,6 +48,11 @@ function OverlayArrow(props: OverlayArrowProps, ref: ForwardedRef<HTMLDivElement
       placement
     }
   });
+  // remove undefined values from renderProps.style object so that it can be
+  // spread merged with the other style object
+  if (renderProps.style) {
+    Object.keys(renderProps.style).forEach(key => renderProps.style![key] === undefined && delete renderProps.style![key]);
+  }
```

Comment:
How did the undefineds get there? I didn't see any style prop in the examples. Will we need this in other places?

Reply:
It just comes from here: https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/overlays/src/calculatePosition.ts#L384
and then it's reinforced here
https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/overlays/src/useOverlayPosition.ts#L224
so even if you removed it from the first one, it'd reappear there.

Now that we don't use mergeProps for everything in RAC, we will need to be careful about this.
Do you have any thoughts or preferences? I am happy to fix it upstream instead. I was just going for smallest change here.

Reply:
mergeProps doesn't merge styles either, so would have to be a new util I guess. fine for now